app/controllers/comments_controller.rb

```diff
@@ -32,12 +36,16 @@ def destroy
   end
 
   def create
-    @comment = @idea.comments.build(comment_params)
-    @comment.user = current_user
-    if @comment.save
-      redirect_to idea_comment_path(@comment.idea, @comment), notice: 'Comment created'
+    if @idea.approved?
```

Comment:
Here you could use the pattern I demonstrated on the `new` method to avoid the lengthy if..else and getting rid of the nesting. :)

Reply:
Done 👍